components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2762,30 +2762,51 @@ private boolean isFilteringNotSupported(String filterOperation) {
     }
 
     /**
-     * get the search value after appending the delimiters
-     * @param filterOperation operator value
-     * @param attributeValue search value
-     * @param delimiter delimiter for based on search type
-     * @return search attribute
+     * Get the search value after appending the delimiters according to the attribute name to be filtered.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Search value
+     * @param delimiter       Filter delimiter based on search type
+     * @return Search attribute
      */
-    private String getSearchAttribute(String filterOperation, String attributeValue, String delimiter) {
+    private String getSearchAttribute(String attributeName, String filterOperation, String attributeValue,
+            String delimiter) {
 
         String searchAttribute = null;
         if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.CO)) {
-            String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR);
-            if(attributeItems.length == 2) {
-                searchAttribute = attributeItems[0] + CarbonConstants.DOMAIN_SEPARATOR +
-                        delimiter + attributeItems[1] + delimiter;
+
+            // For attributes which support domain embedding, create search value by appending the delimiter after
+            // the domain separator.
+            if (isDomainEmbeddingSupportedAttribute(attributeName)) {
+
+                // Check whether domain is embedded in the attribute value.
+                String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR);
+                if (attributeItems.length == 2) {
+                    searchAttribute =
+                            attributeItems[0] + CarbonConstants.DOMAIN_SEPARATOR + delimiter + attributeItems[1]
+                                    + delimiter;
+                } else {
+                    searchAttribute = delimiter + attributeValue + delimiter;
+                }
             } else {
                 searchAttribute = delimiter + attributeValue + delimiter;
             }
         } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.SW)) {
```

Comment:
Why are we treating differently for `SW`, given that for other operations we allow usage with the prepended domain

Reply:
The issue is where the delimiter is appended. In `SW` delimiter is appended only at the end (eg: test/use*). Does not effect when the domain name is in the beginning. But for `CO` delimiter needs to append from both ends (eg: *user1*). So when domain is in front the search value should be as TEST/*1*.